### DISCUSSION TITLE
'Sanic' object has no attribute 'register_listener'

### DISCUSSION THREAD

**Discussion by g1i1U** [idom-team/idom#674] [Problem]
>>> import idom
>>> from idom import component, html, run
>>> 
>>> 
>>> @component
... def App():
...     return html.h1("Hello, ddWorld!")
... 
>>> 
>>> run(App)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/idom/server/prefab.py", line 58, in run
    server = server_type(component, server_config, app)
  File "/usr/local/lib/python3.9/site-packages/idom/server/sanic.py", line 74, in PerClientStateServer
    return SanicServer(app)
  File "/usr/local/lib/python3.9/site-packages/idom/server/sanic.py", line 111, in __init__
    app.register_listener(self._server_did_start, "after_server_start")
AttributeError: 'Sanic' object has no attribute 'register_listener'

**Comment by rmorshea** [comment]
what version of Sanic are you using?

**Comment by g1i1U** [comment]
Name: sanic Version: 18.12.0
Python 3.9.10

**Reply by rmorshea** [reply]
We need to correct the dependencies. IDOM requires sanic>=19,<19.12 which, admittedly, is a rather slim range of versions. This is because of [this issue](https://github.com/idom-team/idom/issues/591).

I'd suggest using a different server implementation if you need more flexibility here (e.g. FastAPI or Starlette) - see docs for [more info](https://idom-docs.herokuapp.com/docs/getting-started/running-idom.html#choosing-a-server-implementation) on using alternate server types.

**Reply by rmorshea** [reply]
Here's an issue which will track the dependency update: https://github.com/idom-team/idom/issues/678
